adobe/spectrum-css
# feat(tooltip): S2 migration

## Description

This migration includes updated colors, rounding, a bigger tip, and the removal of variants (only neutral is available in Spectrum 2). As a result of the deprecation of variants, icons have also been removed.

The redesign of the tip, specifically the rounding, required a reworking of how we use clip-path and transform.

Some custom property mods have been removed:

- `--mod-tooltip-background-color-informative`
- `--mod-tooltip-background-color-negative`
- `--mod-tooltip-background-color-positive`
- `--mod-tooltip-icon-spacing-block-start`
- `--mod-tooltip-icon-spacing-inline-end`
- `--mod-tooltip-icon-spacing-inline-start`
- `--mod-tooltip-icon-width`

And one mod has been added:

- `--mod-tooltip-tip-corner-radius`

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [ ] Design review (@mdt2)
  - Specifically check on the new distance of tooltip from edge. This needed adjusted because of the new corner rounding. It looks funky when the tooltip text is only one line.
- [x] Check out Tooltip in Storybook, specifically the [testing preview](https://pr-2743--spectrum-css.netlify.app/preview/?path=/story/components-tooltip--default&globals=testingPreview:!true) will be helpful. The variants should align with the token changes in the [design spec](https://www.figma.com/file/eoZHKJH9a3LJkHYCGt60Vb/S2-token-specs?type=design&node-id=9602-6409&mode=design&t=TLl0bmGBfKk31oG6-0) (@rise-erpelding)
- [x] Now turn on RTL. All the tips should be in the correct place. (@rise-erpelding)
- [x] Any controls for the deprecated variants (informative, positive, negative) have been removed from storybook (@rise-erpelding)
- [x] Now let's spin up the docs site to check that the "show on hover" functionality is still looking good. You'll have to do this locally because of the difference in tokens package:
  - First update the `site/package.json` tokens version to be `>=14.0.0-next.7`
  - Then you can run a `yarn install`
  - And finally a `yarn dev`
  - Check on that hover functionality and make sure it's looking good
- [x] And still in the docs site, turn on RTL and make sure the hover functionality looks good there too

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:


2. If components have been modified, VRTs have been run on this branch:


## Screenshots

![Screen Shot 2024-05-09 at 3 29 34 PM](https://github.com/adobe/spectrum-css/assets/23404383/7715c921-0b08-4650-b5d9-c9f4595be270)


## To-do list
